### ISSUE TITLE
bootloader, u-boot, and kernel source files

### ISSUE THREAD

**Issue by Kelador** [trimui/firmware_smartpro#29]
Please publish the source files for the bootloader, u-boot, and kernel to help with custom firmware development.

**Comment by jmjoy** [comment]
I think it is unlikely, because this is a "closed source handheld".

**Comment by Kelador** [comment]
> I think it is unlikely, because this is a "closed source handheld".

We can hope it would make good business sense look at other handhelds that are open source it leads to community development which leads to more sales of the handheld.

**Comment by jmjoy** [comment]
Good idea, but it seems like the repository owner never reads the issue.

**Comment by icodeforyou-dot-net** [comment]
I'd also love to see this being open source. That being said, what are the hurdles of compiling a kernel yourself for this device?

**Comment by fonix232** [comment]
> I think it is unlikely, because this is a "closed source handheld".

That doesn't matter - the kernel alone is GPLv2 licensed and such any binary distribution needs to be accompanied with the source code. So is U-Boot, by the way.

So it's not really a question of the firmware being "closed source" but the fact that TrimUI uses open source software that by license, they need to release the modifications to. It's a legal requirement, and if brought to the attention of courts, can result in their products getting an import ban in most Western countries.
